ruma-api-macros/src/api/request.rs

```diff
@@ -517,80 +517,213 @@ impl Request {
+                        .header(
+                            #ruma_api::exports::http::header::CONTENT_TYPE,
+                            "application/json"
+                        );
+
+                    let mut req_headers = req_builder
+                        .headers_mut()
+                        .expect("`http::RequestBuilder` is in unusable state");
+
+                    #header_kvs
+
+                    let http_request = req_builder.body(#request_body)?;
+
+                    Ok(http_request)
                 }
             }
 
-                            #[automatically_derived]
-                            #[cfg(feature = "server")]
-                            impl #ruma_api::IncomingRequest for #incoming_request_type {
-                                type EndpointError = #error_ty;
-                                type OutgoingResponse = Response;
-
-                                const METADATA: #ruma_api::Metadata = self::METADATA;
-
-                                fn try_from_http_request(
-                                    request: #http::Request<Vec<u8>>
-                                ) -> ::std::result::Result<Self, #ruma_api::error::FromHttpRequestError> {
-                                    #extract_request_path
-                                    #extract_request_query
-                                    #extract_request_headers
-                                    #extract_request_body
-
-                                    Ok(Self {
-                                        #parse_request_path
-                                        #parse_request_query
-                                        #parse_request_headers
-                                        #parse_request_body
-                                    })
+            #[automatically_derived]
+            #[cfg(feature = "server")]
+            impl #ruma_api::IncomingRequest for #incoming_request_type {
+                type EndpointError = #error_ty;
+                type OutgoingResponse = Response;
+
+                const METADATA: #ruma_api::Metadata = self::METADATA;
+
+                fn try_from_http_request(
+                    request: #http::Request<Vec<u8>>
+                ) -> ::std::result::Result<Self, #ruma_api::error::FromHttpRequestError> {
+                    #extract_request_path
+                    #extract_request_query
+                    #extract_request_headers
+                    #extract_request_body
+
+                    Ok(Self {
+                        #parse_request_path
+                        #parse_request_query
+                        #parse_request_headers
+                        #parse_request_body
+                    })
+                }
+            }
+        }
+    }
+
+    /// Deserialize the query string.
+    fn extract_request_query(&self, ruma_api: &TokenStream) -> TokenStream {
+        let ruma_serde = quote! { #ruma_api::exports::ruma_serde };
+
+        let request = self;
```

Comment:
This is not necessary. Just replace `request` by `self` below, except inside `quote! {}`.

Reply:
Same for the other `let request = self;` assignments.